joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 25

swerve_controller:
  type: position_controllers/JointPositionController
  joint: swerve_joint
  pid: {p: 1.0, i: 0, d: 0}
  
wheel_controller:
  type: velocity_controllers/JointVelocityController
  joint: wheel_joint
  pid: {p: 5.0, i: 0.01, d: 0.1}

joint_limits:
  swerve_joint:
    has_position_limits: true
    min_position: -3.1415926
    max_position: 3.1415926
    has_velocity_limits: true
    max_velocity: 3.2
    has_acceleration_limits: true
    max_acceleration: 1
  wheel_joint:
    has_position_limits: true
    min_position: -3.1415926
    max_position: 3.1415926
    has_velocity_limits: true
    max_velocity: 12.7
    has_acceleration_limits: true
    max_acceleration: 30